### PR TITLE
:sparkles: changing the default timezone to Europe/Berlin

### PR DESCRIPTION
**Pull request by ChrisB9**
It's not yet entirely feature complete:
- this approach currently seems to remove the possibility to change the timezone via docker-compose

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -54,7 +54,15 @@ ENV \
     PHP_DISPLAY_ERRORS="1" \
     PHP_MEMORY_LIMIT="-1" \
     XHGUI_MONGO_URI="global-xhgui:27017" \
-    XHGUI_PROFILING="enabled"
+    XHGUI_PROFILING="enabled" \
+    TZ=Europe/Berlin \
+    PHP_DATE_TIMEZONE=${TZ} \
+    SET_CONTAINER_TIMEZONE=true \
```
Comment (Kanti) [inline Dockerfile]: needed?

File: Dockerfile [hunk 2]
```diff
@@ -54,7 +54,15 @@ ENV \
...
+    CONTAINER_TIMEZONE=${TZ}
```
Comment (Kanti) [inline Dockerfile]: needed?

File: Dockerfile [hunk 3]
```diff
@@ -54,7 +54,15 @@ ENV \
     PHP_DISPLAY_ERRORS="1" \
     PHP_MEMORY_LIMIT="-1" \
     XHGUI_MONGO_URI="global-xhgui:27017" \
-    XHGUI_PROFILING="enabled"
+    XHGUI_PROFILING="enabled" \
+    TZ=Europe/Berlin \
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
    TIMEZONE=Europe/Berlin \
```

File: Dockerfile [hunk 4]
```diff
@@ -54,7 +54,15 @@ ENV \
...
+    PHP_DATE_TIMEZONE=${TZ} \
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
    PHP_DATE_TIMEZONE=${TIMEZONE} \
```

File: Dockerfile [hunk 5]
```diff
@@ -54,7 +54,15 @@ ENV \
...
+    SET_CONTAINER_TIMEZONE=true \
+    CONTAINER_TIMEZONE=${TZ}
+
+RUN echo ${TZ} >/etc/timezone && \
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
RUN echo ${TIMEZONE} >/etc/timezone && \
```

File: Dockerfile [hunk 6]
```diff
@@ -54,7 +54,15 @@ ENV \
...
+    ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime && \
```
Comment (Kanti) [inline Dockerfile]:
```suggestion
    ln -sf /usr/share/zoneinfo/${TIMEZONE} /etc/localtime && \
```

File: Dockerfile (hunk 3 above)
Comment (ChrisB9) [reply Dockerfile]: no, TZ is a docker recognized variable

File: Dockerfile (hunk 1 above)
Comment (ChrisB9) [reply Dockerfile]: yes, sadly

File: Dockerfile (hunk 2 above)
Comment (ChrisB9) [reply Dockerfile]: yes, sadly

File: Dockerfile (hunk 3 above)
Comment (ChrisB9) [reply Dockerfile]: https://medium.com/developer-space/be-careful-while-playing-docker-about-timezone-configuration-e7a2217e9b76

File: Dockerfile (hunk 1 above)
Comment (Kanti) [reply Dockerfile]: where?

File: Dockerfile (hunk 2 above)
Comment (Kanti) [reply Dockerfile]: where?

File: Dockerfile (hunk 1 above)
Comment (ChrisB9) [reply Dockerfile]: https://www.ivankrizsan.se/2015/10/31/time-in-docker-containers/
using those two env's in production works flawless (without having to change any Dockerfile)

Comment (Kanti) [reply Dockerfile]: only if you add this to your Container:
image
What we do not do

File: Dockerfile (hunk 5 above)
Comment (Kanti) [inline Dockerfile]: maybe we make it a configurable environment variable

File: Dockerfile (hunk 2 above)
Comment (Kanti) [reply Dockerfile]: remove

File: Dockerfile (hunk 1 above)
Comment (Kanti) [reply Dockerfile]: remove

Comment (ChrisB9) [reply Dockerfile]:
```suggestion
```

File: Dockerfile (hunk 2 above)
Comment (ChrisB9) [reply Dockerfile]:
```suggestion
```

File: Dockerfile (hunk 5 above)
Comment (Kanti) [reply Dockerfile]: at startup we should use the TZ variable and run this script

Comment (ChrisB9) [reply Dockerfile]: isnt this way easier?

**Pull request merged**